scripts/mempool.js

```diff
@@ -1,381 +1,391 @@
+            console.log('Started recent tx fetch: ');
+            for (const tx of txs) {
+                // Do not accept any tx which is below the syncHeight
+                if (this.#syncHeight > tx.blockHeight) {
+                    continue;
+                }
+                if (
+                    !this.txmap.has(tx.txid) ||
+                    !this.txmap.get(tx.txid).isConfirmed()
+                ) {
+                    const fullTx = this.parseTransaction(
+                        await getNetwork().getTxFullInfo(tx.txid)
+                    );
+                    await this.updateMempool(fullTx);
                 }
             }
-            this.UTXOs.push(newUTXO);
-        }
-
-        // Re-render the Balance UIs
-        getBalance(true);
-        getStakingBalance(true);
-    }
-
-    /**
-     * Update an existing UTXO, by confirming its pending status
-     * The UTXO must be in
-     * @param {Object} UTXO - Object to be deconstructed
-     * @param {String} UTXO.id - Transaction id
-     * @param {Number} UTXO.vout - vout
-     */
-    updateUTXO({ id, vout }) {
-        if (debug) {
-            console.assert(
-                this.isAlreadyStored({ id, vout }),
-                'Debug Mode: updateUTXO must be called with an existing UTXO'
+            const endTime = new Date();
+            console.log(
+                'Ended recent tx fetch in:',
+                (endTime - startTime) / 1000
             );
-        }
-        const cUTXO = this.UTXOs.find(
-            (utxo) => utxo.id === id && utxo.vout == vout
-        );
-        switch (cUTXO.status) {
-            case Mempool.PENDING:
-                cUTXO.status = Mempool.CONFIRMED;
-                break;
-        }
-        getBalance(true);
-        getStakingBalance(true);
+        });
     }
-
     /**
-     * Remove a UTXO completely from our wallet
-     * @param {UTXO} cUTXO - UTXO to remove
+     * An Outpoint to check
+     * @param {COutpoint} op
      */
-    removeUTXO(cUTXO) {
-        this.UTXOs = this.UTXOs.filter((utxo) => !utxo.equalsUTXO(cUTXO));
+    isSpent(op) {
+        return this.spent.get(op.txid)?.some((x) => x.n == op.n);
     }
 
     /**
-     * Remove a UTXO completely from our wallet, with a 12 minute delay given his id, path and vout
-     * @param {Object} UTXO
-     * @param {String} UTXO.id - Transaction ID
-     * @param {Number} UTXO.vout - Output position of this transaction
+     * Get the total wallet balance
      */
-    autoRemoveUTXO({ id, vout }) {
-        for (const cUTXO of this.UTXOs) {
-            // Loop given + internal UTXOs to find a match, then start the delayed removal
-            if (cUTXO.id === id && cUTXO.vout === vout) {
-                cUTXO.status = Mempool.REMOVED;
-                this.removeWithDelay(12, cUTXO);
-                return;
+    async getBalanceNew(filter) {
+        const startTime = new Date();
+        console.log('Starting calculating total balance');
```

Comment:
```suggestion
```

Reply:
I will remove those logs once the mempool is tested